docs/src/main/paradox/jdbc.md

```diff
@@ -1,8 +1,246 @@
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #grouped-handler }
+
+### Stateful handler
+
+The `JdbcHandler` can be stateful, with variables and mutable data structures. It is invoked by the `Projection` machinery
+one envelope at a time and visibility guarantees between the invocations are handled automatically, i.e. no volatile
+or other concurrency primitives are needed for managing the state as long as it's not accessed by other threads
+than the one that called `process`.
+
+@@@ note
+
+It is important that the `Handler` instance is not shared between several `Projection` instances,
+because then it would be invoked concurrently, which is not how it is intended to be used. Each `Projection`
+instance should use a new `Handler` instance.  
+
+@@@
+
+### Async handler
+
+The @apidoc[Handler] can be used with `JdbcProjection.atLeastOnceAsync` and 
+`JdbcProjection.groupedWithinAsync` if the handler is not storing the projection result in the database.
+The handler could @ref:[send to a Kafka topic](kafka.md#sending-to-kafka) or integrate with something else.
+
+There are several examples of such `Handler` in the @ref:[documentation for Cassandra Projections](cassandra.md#handler).
+Same type of handlers can be used with `JdbcProjection` instead of `CassandraProjection`.
+
+### Actor handler
+
+A good alternative for advanced state management is to implement the handler as an [actor](https://doc.akka.io/docs/akka/current/typed/actors.html),
+which is described in @ref:[Processing with Actor](actor.md).
+
+### Flow handler
+
+An Akka Streams `FlowWithContext` can be used instead of a handler for processing the envelopes,
+which is described in @ref:[Processing with Akka Streams](flow.md).
+
+### Handler lifecycle
+
+You can override the `start` and `stop` methods of the @apidoc[JdbcHandler] to implement initialization
+before first envelope is processed and resource cleanup when the projection is stopped.
+Those methods are also called when the `Projection` is restarted after failure.
+
+See also @ref:[error handling](error.md).
+
+## Schema
+
+The database schema for the offset storage table:
+
+```sql
+create table if not exists "AKKA_PROJECTION_OFFSET_STORE" (
+  "PROJECTION_NAME" VARCHAR(255) NOT NULL,
+  "PROJECTION_KEY" VARCHAR(255) NOT NULL,
+  "OFFSET" VARCHAR(255) NOT NULL,
+  "MANIFEST" VARCHAR(4) NOT NULL,
+  "MERGEABLE" BOOLEAN NOT NULL,
+  "LAST_UPDATED" TIMESTAMP(9) WITH TIME ZONE NOT NULL);
+
+create index "PROJECTION_NAME_INDEX" on "AKKA_PROJECTION_OFFSET_STORE" ("PROJECTION_NAME");
+
+alter table "AKKA_PROJECTION_OFFSET_STORE" 
+    add constraint "PK_PROJECTION_ID" primary key("PROJECTION_NAME","PROJECTION_KEY");
+```
+
+
+@@@ note
+The DDL may need to be adapted according to the specifics of your database brand.
```

Comment:
We could show the ddl for the dialects that we support in separate tabs above. Could also be extracted to a .scala and included with `snip` so that we show the same for jdbc and slick.

Reply:
Yep, I was planing to do that already. But maybe on another PR.

Reply:
Btw, we should also mention that there is the createIfNotExist method on each projection. Useful for tests, but people can also use it to discover how the schema looks like.

Reply:
I added here. No new PR